client/app/src/actions/bookAction.js

```diff
@@ -0,0 +1,208 @@
+import axios from 'axios';
+import {
+  GET_ALL_BOOKS,
+  GET_ALL_BOOKS_ERROR,
+  BORROW_A_BOOK,
+  BORROW_A_BOOK_ERROR,
+  GET_USER_HISTORY,
+  GET_USER_HISTORY_ERROR,
+  RETURN_A_BOOK,
+  RETURN_A_BOOK_ERROR,
+  ADD_BOOK,
+  ADD_BOOK_ERROR,
+  DELETE_BOOK,
+  DELETE_BOOK_ERROR
+} from '../constants/actionTypes';
+
+/**
+ *
+ * @param {msg} message - dispatched message object
+ */
+const deleteBook = deleteMessage => ({
+  type: DELETE_BOOK,
+  deleteMessage
+});
+/**
+ *
+ * @param {error} error - dispatched error object
+ */
+const deleteBookError = error => ({
+  type: DELETE_BOOK_ERROR,
+  error
+});
+/**
+ *
+ * @return {object} All books
+ * @param {error} books - dispatched book object
+ */
+const getAllBooks = books => ({
+  type: GET_ALL_BOOKS,
+  books
+});
+/**
+ *
+ * @return {object} error
+ * @param {error} error - dispatched error object
+ */
+const getAllBooksError = error => ({
+  type: GET_ALL_BOOKS_ERROR,
+  error
+});
+/**
+ *
+ * @return {object} All books
+ * @param {error} books - dispatched book object
+ */
+const borrowBook = book => ({
+  type: BORROW_A_BOOK,
+  book
+});
+
+const borrowBookError = error => ({
+  type: BORROW_A_BOOK_ERROR,
+  error
+});
+/**
+ *
+ * @return {object} All history
+ * @param {error} detailedHistory - dispatched history object
+ */
+const getHistory = detailedHistory => ({
+  type: GET_USER_HISTORY,
+  detailedHistory
+});
+/**
+ *
+ * @return {object} error
+ * @param {error} error - dispatched error object
+ */
+const getHistoryError = error => ({
+  type: GET_USER_HISTORY_ERROR,
+  error
+});
+/**
+ *
+ * @return {object} return book details
+ * @param {error} error - dispatched error object
+ */
+const returnBook = returnMessage => ({
+  type: RETURN_A_BOOK,
+  returnMessage
+});
+/**
+ *
+ * @return {object} message on success
+ * @param {error} message - dispatched error object
+ */
+const addBook = message => ({
+  type: ADD_BOOK,
+  message
+});
+/**
+ *
+ * @return {object} error
+ * @param {error} error - dispatched error object
+ */
+const addBookError = error => ({
+  type: ADD_BOOK_ERROR,
```

Comment:
Nice use of shorthand object. Also note that [shorthand properties should always come first in your object](http://airbnb.io/javascript/#objects--grouped-shorthand) . Please endeavor to apply this guide everywhere in your codebase